criteria.lua

```diff
@@ -43,6 +53,42 @@ local demo_steps = {
   }
 }
 
+-- ---------------------------------------------------------------------------------------------------------------------
+local function create_enemy_forces_bar(step_index)
+  if enemy_forces_bar then
+    return enemy_forces_bar
+  end
+
+  -- frame
+  enemy_forces_bar = CreateFrame("STATUSBAR", nil, main.get_frame())
+  enemy_forces_bar:SetPoint("TOPLEFT", step_frames[step_index - 1], "BOTTOMLEFT", 0, -5)
```

Comment:
I thought this code will handle behaviour for when in one dungeon you have 4 bosses and in next 5.

Reply:
the frame is never deleted between runs, so line 59 will return the "old" frame and that code is never called :) 
the step frames are just hidden after the run, because creating frames is "expensive"